File: app/src/main/java/com/example/animabuilder/CharacteristicInput.java
package com.example.animabuilder;

import android.content.Context;
import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;
import android.widget.TextView;

import com.example.animabuilder.CharacterCreation.BaseCharacter;

import java.util.function.Consumer;
import java.util.function.Supplier;

public class CharacteristicInput implements TextWatcher {

    EditText userInput;
    TextView modOutput;
    Consumer<Integer> valSetter;
    Supplier<Integer> valGetter;
    Context pageOf;

    public CharacteristicInput(EditText userInput, TextView modOutput, Consumer<Integer> valSetter,
                               Supplier<Integer> valGetter, Context pageOf){
        this.userInput = userInput;
        this.modOutput = modOutput;
        this.valSetter = valSetter;
        this.valGetter = valGetter;
        this.pageOf = pageOf;
    }

    @Override
    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void afterTextChanged(Editable editable) {
        applyModVal();
    }

    /**
     * Sets the character's indicated value to the input value and alters the indicated mod display
     * to the corresponding value
     */
    private void applyModVal(){

        //if integer value given
        if(!userInput.getText().toString().equals("")){
            //chanage the indicated value in the character page
            valSetter.accept(Integer.parseInt(userInput.getText().toString()));

            //get the indicated value from the character page
            modOutput.setText(pageOf.getString(R.string.intItem, valGetter.get()));
        }
        //clear mod if text is empty
        else
            modOutput.setText("");
    }
}

File: app/src/main/java/com/example/animabuilder/ListenerImplementations/CharacteristicInput.kt
package com.example.animabuilder.ListenerImplementations

import android.content.Context
import android.widget.EditText
import android.widget.TextView
import android.text.TextWatcher
import android.text.Editable
import com.example.animabuilder.R
import java.util.function.Consumer
import java.util.function.Supplier

class CharacteristicInput(
    var userInput: EditText, var modOutput: TextView, var valSetter: Consumer<Int>,
    var valGetter: Supplier<Int>, var pageOf: Context
) : TextWatcher {
    override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
    override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
    override fun afterTextChanged(editable: Editable) {
        applyModVal()
    }

    /**
     * Sets the character's indicated value to the input value and alters the indicated mod display
     * to the corresponding value
     */
    private fun applyModVal() {

        //if integer value given
        if (userInput.text.toString() != "") {
            //chanage the indicated value in the character page
            valSetter.accept(userInput.text.toString().toInt())

            //get the indicated value from the character page
            modOutput.text = pageOf.getString(R.string.intItem, valGetter.get())
        } else modOutput.text = ""
    }
}